test/HexFloat.cpp

```diff
@@ -1113,11 +1119,118 @@ INSTANTIATE_TEST_CASE_P(
 
         // When negate_value is true, we should not accept a leading minus or
         // plus.
-        BadFloatParseCase<Float16>("-0.0", true),
-        BadFloatParseCase<Float16>("-2.0", true),
-        BadFloatParseCase<Float16>("+0.0", true),
-        BadFloatParseCase<Float16>("+2.0", true),
+        BadFloatParseCase<Float16>("-0.0", true, uint16_t{0}),
+        BadFloatParseCase<Float16>("-2.0", true, uint16_t{0}),
+        BadFloatParseCase<Float16>("+0.0", true, uint16_t{0}),
+        BadFloatParseCase<Float16>("+2.0", true, uint16_t{0}),
     }));
 
-// TODO(awoloszyn): Add fp16 tests and HexFloatTraits.
+// A test case for detecting infinities.
+template <typename T>
+struct InfinityParseCase {
+  std::string input;
+  bool expect_success;
+  T expected_value;
+};
+
+using FloatProxyParseInfinityFloatTest =
```

Comment:
Good point.  Changing Infinity to Overflow in this struct name and test classes and names.